### ISSUE TITLE
[kmac]

### ISSUE THREAD

**Issue by udinator** [lowRISC/opentitan#6635]

**Comment by eunchan** [comment]
Yes. That is expected behavior. Entropy timer runs regardless of whether it is in Key Block or other message blocks. However, even if the timer reaches to 0, it does not trigger the FSM to fetch new entropy from EDN unless it is in key block.

**Comment by udinator** [comment]
Hmm, got it so it is expected to keep counting through fast_entropy keccak hashing.

I'm also seeing the timer stop counting when we are not using fast entropy, as `StRandExpand` does not set `timer_enable = 1` - this is expected behavior as well?

I see, so a timer-related EDN request can only be issued if we're about to process a next keyblock.
Will this happen even if we have just refreshed entropy after a previous hash operation?

**Comment by eunchan** [comment]
> Hmm, got it so it is expected to keep counting through fast_entropy keccak hashing.
> I'm also seeing the timer stop counting when we are not using fast entropy, as StRandExpand does not set timer_enable = 1 - this is expected behavior as well?
> 
Yes. I designed the timer to be decreased in the Ready state.
> I see, so a timer-related EDN request can only be issued if we're about to process a next keyblock.
> Will this happen even if we have just refreshed entropy after a previous hash operation?
> 
It depends on how SW configured the timer value. I expect SW sets long enough timer to cover the scenario. If previous hash operation is done and when new hash starts right after the completion, then it may not initiate new fetch from EDN as timer value is not yet expired. If the next hash operation comes long after the previous completion, then, yes, it will fetch new one.
